ts/src/main/scala/ch/epfl/ts/component/fetch/HistDataCSVFetcher.scala

```diff
@@ -0,0 +1,163 @@
+package ch.epfl.ts.component.fetch
+
+import ch.epfl.ts.data.Quote
+import ch.epfl.ts.data.Currency
+import scala.util.parsing.combinator._
+import scala.io.Source
+import scala.concurrent.duration._
+import java.util.Date
+import java.util.Calendar
+import java.util.Timer
+import java.util.TimerTask
+
+/**
+ * HistDataCSVFetcher class reads data from csv source and converts it to Quotes.
+ * For every Quote it has read from disk, it calls the function callback(q: Quote),
+ * simulating the past by waiting for a certain time t after each call. By default
+ * t is the original (historical) time difference between the quote that was last sent 
+ * and the next quote to be sent.
+ * 
+ * @param dataDir       An absolute directory path. The directory should contain substructures
+ *                      of the form <currency pair>/<xyz>.csv, e.g.: 
```

Comment:
The month is not inferred from the filename, it's inferred from start and end dates. I will clarify that the dataDir cotains *possible* data files you can then chose from using start and end dates.